elements/lisk-p2p/test/unit/p2p_request.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { expect } from 'chai';
+import { P2PRequest, RequestOptions } from '../../src';
+
+describe('p2p_request', () => {
+	let requestOptions: RequestOptions;
+	let respondCallback: (responseError?: Error, responseData?: unknown) => void;
+	let request: P2PRequest;
+
+	beforeEach(() => {
+		requestOptions = {
+			procedure: 'foo',
+			data: 123,
+			id: 'abc123',
+			rate: 0,
+			productivity: {
+				requestCounter: 0,
+				responseCounter: 0,
+				responseRate: 0,
+				lastResponded: 0,
+			},
+		};
+		respondCallback = sandbox.stub();
+		request = new P2PRequest(requestOptions, respondCallback);
+	});
+
+	describe('#constructor', () => {
+		it('should have a procedure property which is set to the value specified in the constructor', async () => {
+			expect(request)
```

Comment:
I think you can just say
`expect(request.procedure).to.equal('foo');`
same for other places